code/modules/mob/living/carbon/carbon_movement.dm

```diff
@@ -2,14 +2,23 @@
 	. = ..()
 	. += grab_state * 3 //can't go fast while grabbing something.
 
-	if(!get_leg_ignore()) //ignore the fact we lack legs
-		var/leg_amount = get_num_legs()
-		. += 6 - 3*leg_amount //the fewer the legs, the slower the mob
-		if(!leg_amount)
-			. += 6 - 3*get_num_arms() //crawling is harder with fewer arms
-
-	if(legcuffed)
-		. += legcuffed.slowdown
+	if(ishuman(src))
+		if(!(dna && dna.species && (FLYING in dna.species.specflags)))		//If someone can get me a better way to do this I'l learn gladly.
```

Comment:
What is the point of this?
get_leg_ignore() already handles this. It literally checks to see if flying is in the specflags.